src/main/kotlin/net/irisshaders/lilybot/extensions/util/ThreadInviter.kt

```diff
@@ -14,51 +14,60 @@ import com.kotlindiscord.kord.extensions.utils.delete
 import com.kotlindiscord.kord.extensions.utils.respond
 import dev.kord.common.entity.ArchiveDuration
 import dev.kord.common.entity.MessageType
-import dev.kord.common.entity.Snowflake
+import dev.kord.core.behavior.channel.MessageChannelBehavior
 import dev.kord.core.behavior.channel.withTyping
 import dev.kord.core.behavior.edit
 import dev.kord.core.behavior.reply
+import dev.kord.core.entity.channel.NewsChannel
 import dev.kord.core.entity.channel.TextChannel
+import dev.kord.core.entity.channel.thread.NewsChannelThread
 import dev.kord.core.entity.channel.thread.TextChannelThread
 import dev.kord.core.event.channel.thread.ThreadChannelCreateEvent
 import dev.kord.core.event.message.MessageCreateEvent
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.last
-import net.irisshaders.lilybot.database.DatabaseHelper
-import net.irisshaders.lilybot.database.DatabaseManager
+import net.irisshaders.lilybot.utils.DatabaseHelper
 import kotlin.time.Duration.Companion.seconds
 import kotlin.time.ExperimentalTime
 
 class ThreadInviter : Extension() {
-	override val name = "threadinviter"
+	override val name = "thread-inviter"
 
 	override suspend fun setup() {
 		/**
 		 * Thread inviting system for Support Channels
 		 * @author IMS212
 		 */
 		event<MessageCreateEvent> {
-			check { failIf(event.message.type == MessageType.ChatInputCommand) } // Don't try to create if the message is a slash command
-			check { failIf(event.message.type == MessageType.ThreadCreated || event.message.type == MessageType.ThreadStarterMessage) } // Don't try and run this if the thread is manually created
-			check { failIf(event.message.author?.id == kord.selfId) }
+			// Don't try to create if the message is in DMs
+			check { failIf { event.guildId == null } }
+			// Don't try to create if the message is a slash command
+			check { failIf { event.message.type == MessageType.ChatInputCommand } }
+			// Don't try and run this if the thread is manually created
+			check { failIf { event.message.type == MessageType.ThreadCreated
+					|| event.message.type == MessageType.ThreadStarterMessage } }
+			// Don't try and create if Lily or another bot sent the message
+			check { failIf { event.message.author?.id == kord.selfId || event.message.author?.isBot == true } }
+			// Don't try to create if the message is already in a thread
+			check { failIf { event.message.getChannel() is TextChannelThread } }
+			// Don't try to create if the message is in an announcements channel
+			check { failIf { event.message.getChannel() is NewsChannel
+					|| event.message.getChannel() is NewsChannelThread } }
```

Comment:
Is this needed this way? Couldn't we check for the opposite? i.e. "if user send message in guild" etc.